sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -875,6 +899,65 @@ private[hive] case class MetastoreRelation
     }
   }
 
+  def prepareStats(partitionPruningPred: Seq[Expression]) {
```

Comment:
Instead of changing the internal state for the case class `MetastoreRelation`, I am wondering if we can add the `predicates` as the attributes of the `MetastoreRelation`.

And then the `lazy val statistics` can get the size info directly by using the code like below.